api/src/services/JobQueue.ts

```diff
@@ -1,16 +1,59 @@
-//import { Queue } from 'bullmq';
-//import { Worker } from 'bullmq';
+// import { Queue } from 'bullmq';
+import { Worker } from 'bullmq';
+const fs = require('fs');
+
+import ImageFile from '../models/ImageFile';
+import JobMetadata from '../models/JobMetadata';
+import PageOrder from '../models/PageOrder';
 
 class JobQueue {
+    // TODO: Type
+    workerFuncs: any = {};
+    manager: Worker;
+
+    start() {
+        this.workerFuncs.derivatives = async job => {
+            console.log(": build derivatives: " + job.data.dir);
+
+            // For each page
+            let order = PageOrder.fromJob(job.data);
```

Comment:
I think in the long term it's going to make more sense to have a src/jobs directory with a separate class for each job type... and then this code could move into a DerivativesJob class with a perform() or work() method... and then the JobQueue can just have a registry of job names to worker classes. I think that will greatly improve readability and maintenance as we get more jobs.